### DISCUSSION TITLE
Managing Tasks with SnowDDL

### DISCUSSION THREAD

**Discussion by leopasta-enable** [littleK0i/SnowDDL#137] [General]
Hi folks,
I am curious to know how other users of snowDDL manages task deployment? Do you suspend all tasks before deploying (in case snowDDL detects a change and needs to apply ALTER TASK)? 
How do you know which ones to resume afterwards?

Interested in what patterns folks have used in the field

**Comment by littleK0i** [comment]
For serious deployments general recommendation is to avoid tasks in all cases and use modern orchestrator instead, e.g. Dagster. It helps to see a full picture and build super lineage across all data sources, transformations and reporing. Tasks are limited to Snowflake only.

For basic deployments it seems like a simple approach works well:

1. Let SnowDDL create & change stuff. Tasks are REPLACED in most cases, not ALTERED, so suspension should not be required.
2. After running "apply" and finishing all other manual SQL queries, run `SHOW TASKS` and find suspended ones. Manually "resume" such tasks. You may also trigger these tasks immediatelly.

You may write a procedure or a script to automate it.

It works well if number of tasks is relatively low (up to 50-100), if tasks are simple and if you do not have complex DAGs with tons of branching. When you start noticing complex DAGs - go for orchestrator as soon as possible.
